acceptance/tests/puppet_device_test.rb

```diff
@@ -0,0 +1,63 @@
+test_name "puppet device is able to run and configure a node"
+
+codedir = "#{master.puppet['codedir']}/modules"
+confdir = master.puppet['confdir']
+common_options="--config /tmp/puppet.conf --debug --trace -w0 --server #{master}"
+
+on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
+on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
+on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
+
+apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
+# create and configure test user to trigger PUP-9642
+user {
+  'test':
+    ensure => 'present',
+}
+
+# testfile for --apply
+file {
+  '/tmp/spinner.pp':
+    content => "spinner { '999': ensure => 'absent' }"
+}
+
+# install and configure the Resource API
+service { 'puppetserver': }
+include 'resource_api::server', 'resource_api::agent'
+MANIFEST
+
+on(master, "cp -v #{confdir}/puppet.conf /tmp/puppet.conf")
+on(master, "echo \"#{<<-CONF}\" >> /tmp/puppet.conf")
+[main]
+user = test
+CONF
```

Comment:
I don't like that this causes `puppet device` to change many directories which are shared with puppetserver to being owned by the `test` user, and never reset back to `puppet`:

```
 Debug: /File[/opt/puppetlabs/puppet/cache]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/var/log/puppetlabs/puppet]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/var/run/puppetlabs]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/opt/puppetlabs/puppet/cache/preview]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/certs]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/public_keys]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/certificate_requests]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/private_keys]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/private]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/certs/fe1y7c56y6cedw2.delivery.puppetlabs.net.pem]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/private_keys/fe1y7c56y6cedw2.delivery.puppetlabs.net.pem]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/public_keys/fe1y7c56y6cedw2.delivery.puppetlabs.net.pem]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/certs/ca.pem]/owner: owner changed 'puppet' to 'test'
  Debug: /File[/etc/puppetlabs/puppet/ssl/crl.pem]/owner: owner changed 'puppet' to 'test'
```

I think the only reason this doesn't break later tests is 1) we're not changing the group, so puppetserver running with group `puppet` can still access the private key, etc, and 2) the PR testing didn't run `with servertests` so it may not have triggered any issues. Probably best to delete the temp `puppet.conf` and then run `puppet agent -t` in a teardown block to ensure the dir permissions are restored.

Reply:
Yeah just running `puppet agent -t --server ...` should do the trick:

```
# /opt/puppetlabs/puppet/bin/puppet agent -td --server ynx5njjfg3u433d.delivery.puppetlabs.net
...
Debug: /File[/opt/puppetlabs/puppet/cache]/owner: owner changed 'test' to 'puppet'
Debug: /File[/var/log/puppetlabs/puppet]/owner: owner changed 'test' to 'puppet'
Debug: /File[/var/run/puppetlabs]/owner: owner changed 'test' to 'puppet'
Debug: /File[/opt/puppetlabs/puppet/cache/preview]/owner: owner changed 'test' to 'puppet'
```